src/main/app/forms/validation/validators/isCountrySupported.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+import { Country } from 'app/common/country'
+
+import { PostcodeInfoClient, PostcodeInfoResponse } from '@hmcts/postcodeinfo-client'
+import * as config from 'config'
+import { request } from 'client/request'
+
+const postcodeClient = new PostcodeInfoClient(config.get<string>('postcodeLookup.apiKey'), request)
+
+@ValidatorConstraint()
+export class CheckCountryConstraint implements ValidatorConstraintInterface {
+
+  async validate (value: any | string, args?: ValidationArguments): Promise<boolean> {
+    if (value === undefined || value === null || value === '') {
+      return true
+    }
+    const postcodeInfoResponse: PostcodeInfoResponse = await postcodeClient.lookupPostcode(value)
```

Comment:
Plase also add scenario where postcode lookup API returns 500 response to see how validator would behave in such scenario.